### PR TITLE
OSDOCS-317: Clarifications for deployment and dc comparison

### PR DESCRIPTION
**Pull request by bergerhoffer**
@huffmanca Would you mind peer reviewing this since you helped me with the some of the wording?

Preview: http://file.rdu.redhat.com/~ahoffer/2019/OSDOCS-317/applications/deployments/what-deployments-are.html#deployments-comparing-deploymentconfigs-what-deployments-are

### CODE REVIEW

File: modules/deployments-comparing-deploymentconfigs.adoc [hunk 1]
```diff
@@ -5,25 +5,33 @@
...
+recommended to use Deployments unless you need a specific feature or behavior
+provided by DeploymentConfigs.
+
+The following sections go into more detail on the differences between the two
+object types to further help you decide which type to use.
+
+[id='deployments-design-{context}']
+== Design
+
+One important difference between Deployments and DeploymentConfigs is the
```
Comment (bergerhoffer) [inline modules/deployments-comparing-deploymentconfigs.adoc]: @tnozicka Can you see if my attempt here at describing these situations is accurate, or if not, what specifically needs to be changed? Thanks!

File: modules/deployments-comparing-deploymentconfigs.adoc [hunk 2]
```diff
@@ -5,25 +5,33 @@
...
+properties of the link:https://en.wikipedia.org/wiki/CAP_theorem[CAP theorem]
+that each design has chosen. DeploymentConfigs prefer consistency, whereas
```
Comment (tnozicka) [inline modules/deployments-comparing-deploymentconfigs.adoc]: "that each design has chosen" + for the rollout process

(Once a new revision is rolled out they behave the same.)

File: modules/deployments-comparing-deploymentconfigs.adoc [hunk 3]
```diff
@@ -5,25 +5,33 @@
...
+Deployments take availability over consistency.
+
+For DeploymentConfigs, if a node running a deployer Pod goes down, it will
+not get replaced. The process waits until the node comes back online or is
+manually deleted. Manually deleting the node also deletes the corresponding Pod.
+This means that you can not delete the Pod to unstick the rollout, as the
+kubelet is responsible for deleting the associated Pod.
+
+However, Deployments are driven from a controller manager. The controller
```
Comment (tnozicka) [inline modules/deployments-comparing-deploymentconfigs.adoc]: "Deployments rollouts are driven..."

File: modules/deployments-comparing-deploymentconfigs.adoc [hunk 4]
```diff
@@ -5,25 +5,33 @@
 [id='deployments-comparing-deploymentconfigs-{context}']
 = Comparing Deployments and DeploymentConfigs
 
-Because deployment configurations existed in {product-title} prior to
```
Comment (tnozicka) [inline modules/deployments-comparing-deploymentconfigs.adoc]: why don't we use `{product-title}` var anymore?

**Review by tnozicka — COMMENTED** [review]
@bergerhoffer thanks for updating it

nits, lgtm otherwise

File: modules/deployments-comparing-deploymentconfigs.adoc (hunk 4 above)
Comment (bergerhoffer) [reply modules/deployments-comparing-deploymentconfigs.adoc]: Good catch, I should have.

**Comment by huffmanca** [comment]
This looks good to me. Thanks for submitting it!

**Review by huffmanca — APPROVED** [review]

**Comment by bergerhoffer** [comment]
This was QE approved.

**Pull request merged**